DataBiosphere/jade-data-repo
# DR-2145: Test and refactor FileMetadataUtils.findNewDirectoryPaths()

* Through testing, I realized we could use a larger cache size. So, I changed the LRU declaration to use the firestore cache size variable.
* Added three test cases:
1. Test finding new directory paths when cache is large enough to hold all items
2. Test the caching mechanism
3. Test that we are only returning unique new paths 

Suggested changes from https://github.com/DataBiosphere/jade-data-repo/pull/1073:
* Switch from LRU to Map - I think we want to keep this as LRU so that it acts like a cache
* Switch from list to set - (Thanks for catching this one, Phil!) Switching this to set allowed us to make sure we're only returning unique new paths to check. 
* Use computeIfAbsent() - done
